### ISSUE TITLE
Oxy 4.7.1 PHP Warning 'Undefined array key "icon-background-color /icon-padding/icon-size"' when saving template/post in Builder

### ISSUE THREAD

**Issue by maltmann-muc** [soflyy/oxygen-bugs-and-features#3451]
**Environment:** PHP 8.1, WP 6.4.1, Oxy 4.7.1 
**Situation:** In Builder when saving template/post

**File and Function:**
wp-content/plugins/oxygen/component-framework/component-init.php
ct_generate_class_states_css()

**Warning and Call Stack:**
```
Warning:  Undefined array key "icon-background-color" in /.../wp-content/plugins/oxygen/component-framework/component-init.php on line 3024
Warning:  Undefined array key "icon-padding" in /.../wp-content/plugins/oxygen/component-framework/component-init.php on line 3025
Warning:  Undefined array key "icon-size" in /.../wp-content/plugins/oxygen/component-framework/component-init.php on line 3026
Stack trace:
  1. {main}() /.../index.php:0
  2. require() /.../index.php:17
  3. require_once() /.../wp-blog-header.php:19
  4. do_action($hook_name = 'template_redirect') /.../wp-includes/template-loader.php:13
  5. WP_Hook->do_action($args = [0 => '']) /.../wp-includes/plugin.php:517
  6. WP_Hook->apply_filters($value = '', $args = [0 => '']) /.../wp-includes/class-wp-hook.php:348
  7. ct_save_components_tree_as_post('') /.../wp-includes/class-wp-hook.php:324
  8. oxygen_vsb_cache_universal_css() /.../wp-content/plugins/oxygen/component-framework/includes/ajax.php:273
  9. oxygen_vsb_get_classes_styles() /.../wp-content/plugins/oxygen/component-framework/includes/cache.php:19
 10. ct_generate_class_states_css($class = 'aaa-video-control', $state = 'hover', $options = ['icon-color' => '#ffffff'], $is_media = *uninitialized*, $is_selector = *uninitialized*) /.../wp-content/plugins/oxygen/component-framework/component-init.php:2872

```

**Description:**
$options may or may not contain fields "icon-background-color", "icon-padding", "icon-size".
<img width="1818" alt="Bildschirmfoto 2023-11-19 um 10 17 23" src="https://github.com/soflyy/oxygen-bugs-and-features/assets/51136650/7f6c3aef-7c5c-4092-8127-1b91c1bcfc54">

**Solution:**
Change
$iconBackgroundColor= oxygen_vsb_get_global_color_value($options['icon-background-color']);
$iconPadding        = $options['icon-padding'];
$iconSize           = $options['icon-size'];
To
$iconBackgroundColor= oxygen_vsb_get_global_color_value($options['icon-background-color']??null);
$iconPadding        = $options['icon-padding']??null;
$iconSize           = $options['icon-size']??null;

**Comment by maltmann-muc** [comment]
Oxy 4.8 RC 1: Issue persists.

**Comment by maltmann-muc** [comment]
Oxy 4.8: Issue persists.

**Comment by Kpudlo** [comment]
Confirmed and reported internally via https://github.com/soflyy/oxygen/issues/5055

**Comment by danner26** [comment]
Seeing this same issue on the latest prod build with 6.5.2

**Comment by maltmann-muc** [comment]
Oxy 4.9: Issue persists.

**Comment by Giulioski80** [comment]
I am still having this problem. How can be solved?

**Comment by lucianahanan** [comment]
Oxygen 4.9 issue still persists...

PHP Warning:  Undefined array key "icon-color" in /home/unplugge/store.unplugged.com/wp-content/plugins/oxygen/component-framework/component-init.php on line 3127
[21-Nov-2024 01:45:56 UTC] PHP Warning:  Undefined array key "icon-background-color" in /home/unplugge/store.unplugged.com/wp-content/plugins/oxygen/component-framework/component-init.php on line 3128
[21-Nov-2024 01:45:56 UTC] PHP Warning:  Undefined array key "icon-padding" in /home/site/wp-content/plugins/oxygen/component-framework/component-init.php on line 3129
[21-Nov-2024 01:45:56 UTC] PHP Warning:  Undefined array key "icon-color" in /home/site/wp-content/plugins/oxygen/component-framework/component-init.php on line 3127
[21-Nov-2024 01:45:56 UTC] PHP Warning:  Undefined array key "icon-background-color" in /home/site/wp-content/plugins/oxygen/component-framework/component-init.php on line 3128
[21-Nov-2024 01:45:56 UTC] PHP Warning:  Undefined array key "icon-padding" in /home/site/wp-content/plugins/oxygen/component-framework/component-init.php on line 3129

**Comment by lucianahanan** [comment]
php 8.1 / Oxygen 4.9 issue still persists... @Kpudlo any news about this?

Undefined array key "icon-color" in .../oxygen/component-framework/component-init.php on line 3127

Undefined array key "icon-background-color" in .../oxygen/component-framework/component-init.php on line 3128

Undefined array key "icon-padding" in .../oxygen/component-framework/component-init.php on line 3129

Undefined array key "icon-color" in .../oxygen/component-framework/component-init.php on line 3127

Undefined array key "icon-background-color" in .../oxygen/component-framework/component-init.php on line 3128

Undefined array key "icon-padding" in .../oxygen/component-framework/component-init.php on line 3129

**Comment by maltmann-muc** [comment]
Finally fixed in 4.9.4
